### PR TITLE
Now we can just use 0.0.0.0 - infered by the libp2p chat example

### PR DESCRIPTION
**Pull request by ericxtang**
This was breaking our connection logic because we were listening on the wrong interface.

### CODE REVIEW

File: network_node.go [hunk 1]
```diff
@@ -3,25 +3,23 @@ package basicnet
...
-	"github.com/golang/glog"
-
-	addrutil "gx/ipfs/QmNSWW3Sb4eju4o2djPQ1L1c2Zj9XN9sMYJL8r1cbxdc6b/go-addr-util"
 	bhost "gx/ipfs/QmNh1kGFFdsPu79KNSaL4NUKUPb4Eiz4KHdMtFY6664RDp/go-libp2p/p2p/host/basic"
 	rhost "gx/ipfs/QmNh1kGFFdsPu79KNSaL4NUKUPb4Eiz4KHdMtFY6664RDp/go-libp2p/p2p/host/routed"
 	host "gx/ipfs/QmNmJZL7FQySMtE2BQuLMuZg2EB2CLEunJJUSVSc9YnnbV/go-libp2p-host"
 	swarm "gx/ipfs/QmSwZMWwFZSUpe5muU2xgTUwppH24KfMwdPXiwbEp2c6G5/go-libp2p-swarm"
 	kb "gx/ipfs/QmTH6VLu3WXfbH3nuLdmscgPWuiPZv3GMJ2YCdzBS5z91T/go-libp2p-kbucket"
-	ma "gx/ipfs/QmWWQ2Txc2c6tqjsBpzg5Ar652cHPGNsQQp2SejkNmkUMb/go-multiaddr"
+	multiaddr "gx/ipfs/QmWWQ2Txc2c6tqjsBpzg5Ar652cHPGNsQQp2SejkNmkUMb/go-multiaddr"
```
Comment (j0sh) [inline network_node.go]: Curious why the change here, `ma` is used rather consistently in the codebase. Personally, I prefer the brevity.

**Review by j0sh — COMMENTED** [review]
I believe we still have to enumerate the addresses for `TranscodeSub`. What interfaces does 0.0.0.0 cover that's missed by the old method of explicit enumeration? That would affect the ability to do direct connections if we're not including the proper IP.

**Comment by j0sh** [comment]
Trying to understand this behavior. I actually checked the bindings earlier today and it only seems to bind to one IP for the interface, whereas AWS gives us an internal /20 per host. Also, I'm not sure which corresponds to host's public IP (if there is even a correspondence). But I'm still not sure how exactly the libp2p update broke that (unless it was listening to 0.0.0.0 under the hood the whole time).

**Comment by ericxtang** [comment]
I'm not sure what changes caused this issue - the only thing I did was try to follow the new libp2p example and isolate the difference.  I'll reach out to the protocol labs team about it.

Btw, `0.0.0.0` apparently breaks out tests because it makes the tests significantly slower to the point that it eventually times out.  So I still use `addrutil.InterfaceAddresses()` for when running tests, but expose the option to use `0.0.0.0` for users of the SDK.

Left a question here for the libp2p team - https://github.com/libp2p/go-addr-util/issues/12

**Comment by j0sh** [comment]
I like the method of passing in the binding multiaddrs explicitly. For the tests we can probably simplify by binding to 127.0.0.1 alone.

Since we still need an explicit address for TranscodeSub, it'd be good to see if the Protocol Labs team has any insight on the change.

**Review by j0sh — APPROVED** [review]
